### ISSUE TITLE
[Bug Report] 我注意到你们上传到Transformer的库中的代码存在一个bug

### ISSUE THREAD

**Issue by teamtee** [QwenLM/Qwen2-Audio#137]
We have detected a bug in the Transformer library you uploaded. Details are as follows:
```
transformer: v4.46.2
file: modeling_qwen2_audio.py
function: _merge_input_ids_with_audio_features
```
The error code starts from line 1046:
```
        # 4. Fill the embeddings based on the mask. If we have ["hey" "<audio>", "how", "are"]
        # we need to index copy on [0, 577, 578, 579] for the text and [1:576] for the audio features
        final_embedding[batch_indices, text_to_overwrite] = inputs_embeds[batch_indices, non_audio_indices]
        final_attention_mask[batch_indices, text_to_overwrite] = attention_mask[batch_indices, non_audio_indices]
        final_input_ids[batch_indices, text_to_overwrite] = input_ids[batch_indices, non_audio_indices]
        final_labels = None
        if labels is not None:
            labels = labels.to(target_device)
            final_labels = torch.full_like(final_attention_mask, self.config.ignore_index).to(torch.long)
            final_labels[batch_indices, text_to_overwrite] = labels[batch_indices, non_audio_indices]
```
The main cause of the error is this line of code:
```
 final_labels = torch.full_like(final_attention_mask, self.config.ignore_index).to(torch.long)
```
`torch.full_like` creates a tensor with the same type as the given tensor. Since `final_attention_mask` is of boolean type, the value `-100` specified by `self.config.ignore_index` is converted to `True`, and then `.to(torch.long)` converts `True` to `1`. As a result, all labels that should be ignored are incorrectly set to `1`.

it should be
```
final_labels = torch.full((batch_size, max_token_num), self.config.ignore_index,dtype=input_ids.dtype, device=inputs_embeds.device).to(torch.long)
```

我们注意的到你们上传到Transformer的库中存在一个bug,具体情况如下
```
transformer: v4.46.2
file: modeling_qwen2_audio.py
function: _merge_input_ids_with_audio_features
```
下面是详细的错误代码,开始于1046行
```
        # 4. Fill the embeddings based on the mask. If we have ["hey" "<audio>", "how", "are"]
        # we need to index copy on [0, 577, 578, 579] for the text and [1:576] for the audio features
        final_embedding[batch_indices, text_to_overwrite] = inputs_embeds[batch_indices, non_audio_indices]
        final_attention_mask[batch_indices, text_to_overwrite] = attention_mask[batch_indices, non_audio_indices]
        final_input_ids[batch_indices, text_to_overwrite] = input_ids[batch_indices, non_audio_indices]
        final_labels = None
        if labels is not None:
            labels = labels.to(target_device)
            final_labels = torch.full_like(final_attention_mask, self.config.ignore_index).to(torch.long)
            final_labels[batch_indices, text_to_overwrite] = labels[batch_indices, non_audio_indices]
```
错误的主要原因在于这个代码
```
 final_labels = torch.full_like(final_attention_mask, self.config.ignore_index).to(torch.long)
```
torch.full_like会按照给定的类型来创建tensor,final_attention_mask的类型为bool，因此self.config.ignore_index给定的-100会被转化为True,而.to(torch.long)会导致True被转化为1，这使得最后的结果中所有的应该被忽略的标签都被转化为1.

正确的做法如下
```
final_labels = torch.full((batch_size, max_token_num), self.config.ignore_index,dtype=input_ids.dtype, device=inputs_embeds.device).to(torch.long)
```
